modules/core/src/webgl/classes/texture-cube.js

```diff
@@ -81,15 +84,30 @@ export default class TextureCube extends Texture {
     const {gl} = this;
     const imageDataMap = pixels || data;
 
-    return Promise.all(FACES.map(face => imageDataMap[face])).then(resolvedFaces => {
+    return Promise.all(
+      FACES.map(face => {
+        const facePixels = imageDataMap[face];
+        return Promise.all(Array.isArray(facePixels) ? facePixels : [facePixels]);
+      })
+    ).then(resolvedFaces => {
       this.bind();
 
       FACES.forEach((face, index) => {
-        if (width && height) {
-          gl.texImage2D(face, 0, format, width, height, border, format, type, resolvedFaces[index]);
-        } else {
-          gl.texImage2D(face, 0, format, format, type, resolvedFaces[index]);
+        if (resolvedFaces[index].length > 1 && this.opts.mipmaps !== false) {
```

Comment:
I think we need some comments here, otherwise this will break in next refactor

Reply:
Probably need unit test for that too.